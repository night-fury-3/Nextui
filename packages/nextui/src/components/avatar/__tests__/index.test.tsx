import React from 'react';
import { mount, render, shallow } from 'enzyme';
import { Avatar } from '@components';

describe('Avatar', () => {
  it('should support square and circle', () => {
    const circle = shallow(<Avatar />);
    expect(() => circle.unmount()).not.toThrow();
    const square = shallow(<Avatar squared />);
    expect(() => square.unmount()).not.toThrow();
  });

  it('should render text element', () => {
    const imageAvatar = render(<Avatar />);
    expect(imageAvatar).toMatchSnapshot();
    const textAvatar = render(<Avatar text="text" />);
    expect(textAvatar).toMatchSnapshot();
  });

  it('should omit long chars automatically', () => {
    const avatar = mount(<Avatar text="loremipsumtextlarge" />);
    const text = avatar.find('.avatar-text').text();
    expect(text.length).toBeLessThan(4);
  });

  it('stacked should be work', () => {
    const avatar = shallow(
      <Avatar src="https://i.pravatar.cc/300?u=a042581f4e29026705d" stacked />
    );
    expect(() => avatar.unmount()).not.toThrow();
  });

  it('should work correctly with different sizes', () => {
    const wrapper = mount(
      <div>
        <Avatar size="mini" />
        <Avatar size="small" />
        <Avatar size="medium" />
        <Avatar size="large" />
        <Avatar size="xlarge" />
      </div>
    );
    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it('should render component of a specified size', () => {
    const avatar = render(<Avatar size={20} />);
    expect(avatar).toMatchSnapshot();
  });

  it('should throw error when avatar color is invalid', () => {
    let errorMessage = '';
    const errorSpy = jest
      .spyOn(console, 'error')
      .mockImplementation((msg) => (errorMessage = msg));
    mount(<Avatar color="#kajsdkl" />);
    expect(errorMessage).toContain('is not a valid color');
    errorSpy.mockRestore();
  });
});
