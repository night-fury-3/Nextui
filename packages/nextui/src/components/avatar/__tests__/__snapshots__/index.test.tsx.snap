// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render component of a specified size 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": "jsx-1634182531 avatar  ",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "jsx-1634182531 avatar-text",
        },
        "children": Array [],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Avatar should render correctly with an icon 1`] = `"<span class=\\"jsx-3671186691 avatar  \\"><span class=\\"jsx-3671186691 icon\\"><svg class=\\"\\"></svg></span></span>"`;

exports[`Avatar should render text element 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": "jsx-3671186691 avatar  ",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "jsx-3671186691 avatar-text",
        },
        "children": Array [],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Avatar should render text element 2`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": "jsx-3671186691 avatar  ",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "jsx-3671186691 avatar-text",
        },
        "children": Array [
          Node {
            "data": "text",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Avatar should work correctly with different sizes 1`] = `"<div><span class=\\"jsx-2879363747 avatar  \\"><span class=\\"jsx-2879363747 avatar-text\\"></span></span><span class=\\"jsx-4278147747 avatar  \\"><span class=\\"jsx-4278147747 avatar-text\\"></span></span><span class=\\"jsx-3671186691 avatar  \\"><span class=\\"jsx-3671186691 avatar-text\\"></span></span><span class=\\"jsx-2079164419 avatar  \\"><span class=\\"jsx-2079164419 avatar-text\\"></span></span><span class=\\"jsx-3155889251 avatar  \\"><span class=\\"jsx-3155889251 avatar-text\\"></span></span></div>"`;

exports[`Avatar should work with a gradient color 1`] = `"<span class=\\"jsx-2519447360 avatar  \\"><span class=\\"jsx-2519447360 avatar-text\\"></span></span>"`;

exports[`Avatar should work with different colors 1`] = `"<div><span class=\\"jsx-2848712817 avatar  \\"><span class=\\"jsx-2848712817 avatar-text\\"></span></span><span class=\\"jsx-2340694469 avatar  \\"><span class=\\"jsx-2340694469 avatar-text\\"></span></span><span class=\\"jsx-3117078237 avatar  \\"><span class=\\"jsx-3117078237 avatar-text\\"></span></span><span class=\\"jsx-561504774 avatar  \\"><span class=\\"jsx-561504774 avatar-text\\"></span></span><span class=\\"jsx-854317330 avatar  \\"><span class=\\"jsx-854317330 avatar-text\\"></span></span></div>"`;

exports[`Avatar should work with pointer 1`] = `"<span class=\\"jsx-2505924119 avatar  \\"><span class=\\"jsx-2505924119 avatar-text\\"></span></span>"`;

exports[`Avatar should work with zoomed 1`] = `"<span class=\\"jsx-986380238 avatar  \\"><span class=\\"jsx-986380238 avatar-text\\"></span></span>"`;
